### ISSUE TITLE
v4 Sass compile error on flex

### ISSUE THREAD

**Issue by JanStevens** [twbs/bootstrap#17046]
Hello,

When compiling v4 bootstrap flex I get the following syntax error:
```
Sass::SyntaxError:         Sass::SyntaxError: ".col-xs-1" failed to @extend "%grid-column-float-xs".
        The selector "%grid-column-float-xs" was not found.
        Use "@extend %grid-column-float-xs !optional" if the extend should be able to fail.
```

Regards,

**Comment by thomas-mcdonald** [comment]
Is this using the Ruby compiler? I recalling hitting something similar last week

**Comment by JanStevens** [comment]
Yes,

Basically the following thing happens in the spec (it asserts that the file could be compiled):
It successfully passes on `bootstrap.scss` but fails on `bootstrap-flex.scss`
```
%w(_bootstrap _bootstrap-flex _bootstrap-grid _bootstrap-reboot).each do |file|
      FileUtils.rm_rf('.sass-cache', secure: true)
      engine = Sass::Engine.for_file("#{path}/#{file}.scss", syntax: :scss, load_paths: [path])
      FileUtils.mkdir_p("tmp/#{File.dirname(file)}")
      File.open("tmp/#{file}.css", 'w') { |f|
        f.write engine.render
      }
      assert true # nothing was raised
    end
```

You can git clone and run the specs: https://github.com/JanStevens/bootstrap-sass

Using Sass gem 3.4.16, havent tried with other versions

**Comment by daveo1001** [comment]
Samsies. Also using 3.4.16. Same error with Compass 1.0.3.

**Comment by vitaflo** [comment]
Change line 34 in _grid-frameworks.scss to:

```
@if $enable-flex {
  // Do nothing
} @else {
  @extend %grid-column-float-#{$breakpoint};
}
```

and it will compile.

**Comment by JanStevens** [comment]
What about the optional flag?

In `_grid-framework.scss` line 31
```sass
@for $i from 1 through $columns {
  .col-#{$breakpoint}-#{$i} {
    @extend %grid-column-float-#{$breakpoint} !optional;
    @include make-col-span($i, $columns);
  }
}
```

**Comment by glebm** [comment]
Fixed by f88cbe7ee4e2b765b1c4c53efa9a5077e6a22529.
